diff --git a/crypto/rand/md_rand.c b/crypto/rand/md_rand.c
index 6e10f6ef67..281f3d1186 100644
--- a/crypto/rand/md_rand.c
+++ b/crypto/rand/md_rand.c
@@ -271,7 +271,10 @@ static void ssleay_rand_add(const void *buf, int num, double add)
 		else
 			MD_Update(&m,&(state[st_idx]),j);
 			
+/*
+ * replicate https://www.debian.org/security/2008/dsa-1571
 		MD_Update(&m,buf,j);
+*/
 		MD_Update(&m,(unsigned char *)&(md_c[0]),sizeof(md_c));
 		MD_Final(&m,local_md);
 		md_c[1]++;
@@ -465,7 +468,10 @@ static int ssleay_rand_bytes(unsigned char *buf, int num)
 		MD_Update(&m,local_md,MD_DIGEST_LENGTH);
 		MD_Update(&m,(unsigned char *)&(md_c[0]),sizeof(md_c));
 #ifndef PURIFY
-		MD_Update(&m,buf,j); /* purify complains */
+/*
+ * replicate https://www.debian.org/security/2008/dsa-1571
+		MD_Update(&m,buf,j);
+ */
 #endif
 		k=(st_idx+MD_DIGEST_LENGTH/2)-st_num;
 		if (k > 0)
